source/modes/GameMode.cpp

```diff
@@ -1313,11 +1406,13 @@ void GameMode::refreshResearchButtonState(ResearchType resType)
     }
     else
     {
+        // The skill is not available and research is not pending
         guiSheet->getChild(ceguiWidgetButtonName)->hide();
         researchButton->setText("");
         researchButton->setProperty("StateImage", "");
         researchButton->setEnabled(true);
     }
+    // TODO: handle case research is disabled
```

Comment:
I just forgot to remove it. The case when research is disabled is already handled